code/mob/living/critter/flock/flockdrone.dm

```diff
@@ -73,6 +69,17 @@
 		src.contexts += new type
 	APPLY_ATOM_PROPERTY(src, PROP_ATOM_FLOCK_THING, src)
 	src.AddComponent(/datum/component/flock_protection, FALSE, FALSE, FALSE, FALSE)
+	src.RegisterSignal(src, COMSIG_MOB_GRABBED, .proc/do_antigrab)
+
+/mob/living/critter/flock/drone/proc/do_antigrab(source, var/obj/item/grab/grab)
+	SPAWN(1.5 SECONDS)
+		if (QDELETED(src) || !isalive(src) || QDELETED(grab) || !grab.affecting || !grab.assailant)
```

Comment:
Regarding the isalive() check, feel like maybe it would be better to just unregister the signal on death, and do a dormant check atm, considering undormantize() isn't used right now and also isn't finished/complete